cowprotocol/cow-amm
# Add mainnet deployments

Adds artifacts from contract deployments on mainnet.

This deployment was dirty. I executed:
```sh
forge script 'script/DeployAllContracts.s.sol:DeployAllContracts' -vvvv --rpc-url "$ETH_RPC_URL" --verify --broadcast --private-key "$PK"
```

But the gas price increased between the time of simulation and the time of execution, and so the third transaction failed, since there weren't enough funds in the deployer to pay for it anymore (`(code: -32010, message: InsufficientFunds, Account balance: 391884345774399636, cumulative cost: 397714356026954136, data: None)`).

Since it would be a waste of ETH to relaunch the full deployment script, I decided to just run the deployment script for the remaining contract once the gas price decreased a bit:
```sh
forge script 'script/single-deployment/BalancerWeightedPoolPriceOracle.s.sol:DeployBalancerWeightedPoolPriceOracle' -vvvv --rpc-url "$ETH_RPC_URL"   --private-key "$PK"
```
 
This is why there are two files in `broadcast/`.

I also updated the script that updates `networks.json` to ignore contract creations without a transaction and used it to generate that file.

### How to test

Check out `networks.json` and confirm that the verified contract code matches the expected contracts.
